### PR TITLE
Introduce a method in `MockHttpServletRequestBuilder` to set remote address

### PR DESCRIPTION
**Pull request by leewin12**
Introduce a method in `MockHttpServletRequestBuilder` to set the remote server address.

This PR was discussed in gh-30484.

### CODE REVIEW

File: spring-test/src/test/java/org/springframework/test/web/servlet/request/MockHttpServletRequestBuilderTests.java [hunk 1]
```diff
@@ -31,7 +31,6 @@
 import jakarta.servlet.ServletContext;
 import jakarta.servlet.http.Cookie;
 import org.junit.jupiter.api.Test;
-
```
Comment (sbrannen) [inline spring-test/src/test/java/org/springframework/test/web/servlet/request/MockHttpServletRequestBuilderTests.java:34]: Please reinstate the blank line.

File: spring-test/src/main/java/org/springframework/test/web/servlet/request/MockHttpServletRequestBuilder.java [hunk 2]
```diff
@@ -526,6 +529,17 @@ public MockHttpServletRequestBuilder principal(Principal principal) {
 		return this;
 	}
 
+	/**
+	 * Set the remote address of the request.
+	 * @param remoteAddress the remote address (IP)
+	 * @Since 6.1.0
+	 */
+	public MockHttpServletRequestBuilder remoteAddress(String remoteAddress) {
+		Assert.notNull(remoteAddress, "'remoteAddress' must not be null");
```
Comment (sbrannen) [inline spring-test/src/main/java/org/springframework/test/web/servlet/request/MockHttpServletRequestBuilder.java:538]:
```suggestion
		Assert.hasText(remoteAddress, "'remoteAddress' must not be null or blank");
```

**Review by sbrannen — CHANGES_REQUESTED** [review]
I think this PR looks pretty good.

I've only requested a few minor changes.

File: spring-test/src/test/java/org/springframework/test/web/servlet/request/MockHttpServletRequestBuilderTests.java (hunk 1 above)
Comment (leewin12) [reply spring-test/src/test/java/org/springframework/test/web/servlet/request/MockHttpServletRequestBuilderTests.java]: Oops, thank you for letting me know. I've fixed it. [6783610](https://github.com/spring-projects/spring-framework/pull/30497/commits/678361086da5e32bfc4de567d8353afe68bfb6a3)

**Review by sbrannen — APPROVED** [review]

**Comment by sbrannen** [comment]
Congratulations on submitting your first PR for the Spring Framework! 👍 

This has been merged into `main`.

Thanks

**Comment by leewin12** [comment]
Thank you, @sbrannen! Could we backport this to 5.x? Spring Boot 2.x still relies on 5.x.

**Comment by sbrannen** [comment]
Reopening to introduce in `6.0.x`.

**Pull request closed without merging**